source/faucet/main.d

```diff
@@ -282,6 +283,71 @@ public class Faucet : IFaucet
                    .map!(kv => tuple(kv.key, kv.value))
                    .assocArray();
     }
+
+    /// POST: /send_transaction
+    public override void sendTransaction (string recv, ulong amount)
+    {
+        PublicKey pubkey = PublicKey.fromString(recv);
+        Amount leftover = Amount(amount);
+        auto first_utxo = this.getOwnedUTXOs().byKeyValue().take(1);
+        // A storage to keep track of used UTXOs
+        UTXO[Hash] used_utxos;
```

Comment:
Shouldn't this be a field in the class?

Reply:
Good idea. Endpoints in the future can have access to it.
Do you think there should be a function that returns the `used_utxos`?

Reply:
That's not my point. 

My point is that you need to track used utxos and remember that the faucet's `sendTransaction` will be called many times.

Imagine if the block time interval is 10 minutes. Now one user uses `sendTransaction` and you use one owned UTXO and you send the transaction. The transaction is not yet externalized.

Now another user calls `sendTransaction`, and what happens? You're going to end up using the same UTXO you did for the first user. This is why you need to keep the used utxos as a field.

Reply:
Got it, thanks for the explanation!